### PR TITLE
Geode-3415: Use SslException with fixed circular dependency

### PR DESCRIPTION
**Pull request by mmartell**

### CODE REVIEW

**Comment by mmartell** [comment]
Just noticed I left some test code in.

File: cryptoimpl/SSLImpl.cpp [hunk 1]
```diff
@@ -21,9 +21,9 @@
 
 #include <ace/Guard_T.h>
 
-#include <geode/ExceptionTypes.hpp>
+//#include <geode/ExceptionTypes.hpp>
```
Comment (pivotal-jbarrett) [inline cryptoimpl/SSLImpl.cpp]: Delete not comment.

Comment (mmartell) [reply cryptoimpl/SSLImpl.cpp]: Good catch. Done.

**Comment by mmartell** [comment]
For context, refer to PR #483: https://github.com/apache/geode-native/pull/483

**Comment by pivotal-jbarrett** [comment]
Never! No! 

`std::exception` isn’t meant to be thrown directly. It lacks a constructors for `what`.

Throw `std::invalid_argument`.

**Comment by mmartell** [comment]
std::invalid_argument seems totally appropriate for passing an invalid password. What about failure to read a certificate? Would std::runtime_error be more appropriate? It allows sets the what member to the string.

**Comment by echobravopapa** [comment]
Is it failure to "read" the certificate; i.e. there was no certificate?  or this is an invalid certificate, which aligns to an invalid_argument...

**Comment by pivotal-jbarrett** [comment]
All failures are a result of the arguments passed in. So `std::invalid_argument` is close enough.

**Pull request merged**